core/query/sql.ml

```diff
@@ -137,138 +152,162 @@ let order_by_clause n =
+  let pr_b ppf q = pr_base ppf quote false q in
+  let pr_b_ignore_fields ppf q = pr_base ppf quote true q in
+
+  let gen_pp_pair ppf fmt_str fl fr (l, r) = Format.fprintf ppf fmt_str fl l fr r in
+  let gen_pp_option ppf fmt_str f option =
+    match option with
+      | None -> ()
+      | Some x -> Format.fprintf ppf fmt_str f x
+  in
+
+  let pr_fields ppf fields =
+    let pp_field ppf (b, l) = gen_pp_pair ppf "(%a) as %a" pr_b pp_quote (b, l) in
     if ignore_fields then
-      "0 as \"@unit@\"" (* SQL doesn't support empty records! *)
+      Format.fprintf ppf "%a" pp_constant "0 as \"@unit@\"" (* SQL doesn't support empty records! *)
     else
       match fields with
-        | [] -> "0 as \"@unit@\"" (* SQL doesn't support empty records! *)
-        | fields ->
-          mapstrcat ","
-            (fun (b, l) ->
-              "(" ^ sb b ^ ") as "^ quote l) (* string_of_label l) *)
-            fields
+        | [] -> Format.fprintf ppf "%a" pp_constant "0 as \"@unit@\"" (* SQL doesn't support empty records! *)
+        | fields -> Format.fprintf ppf "%a" (Format.pp_print_list ~pp_sep:pp_comma pp_field) fields
   in
-  let string_of_select fields tables condition os =
-    let tables = String.concat "," tables in
-    let fields = string_of_fields fields in
-    let orderby =
+
+  let pr_select ppf fields tables condition os =
+    let pp_os_condition ppf a = Format.fprintf ppf "%a" pr_b a in
+    let pp_orderby ppf os =
       match os with
-        | [] -> ""
-        | _ -> " order by " ^ mapstrcat "," sb os in
-    let where =
+        | [] -> ()
+        | _ -> Format.fprintf ppf " order by %a" (Format.pp_print_list ~pp_sep:pp_comma pp_os_condition) os in
+    let pp_where ppf condition =
       match condition with
-        | Constant (Constant.Bool true) -> ""
-        | _ ->  " where " ^ sb condition
-    in
-      "select " ^ fields ^ " from " ^ tables ^ where ^ orderby
-  in
-  let string_of_delete table where =
-    let where =
-      OptionUtils.opt_app
-        (fun x -> "where (" ^ sbt x ^ ")") "" where in
-    Printf.sprintf "delete from %s %s" table where
+        | Constant (Constant.Bool true) -> ()
+        | _ -> Format.fprintf ppf " where %a" pp_os_condition condition in
+
+    Format.fprintf ppf "select %a from %a%a%a"
+      pr_fields fields
+      (Format.pp_print_list ~pp_sep:pp_comma Format.pp_print_string) tables
+      pp_where condition
+      pp_orderby os
   in
-  let string_of_update table fields where =
-    let fields =
-      List.map (fun (k, v) -> quote k ^ " = " ^ sbt v) fields
-      |> String.concat ", " in
 
-    let where =
-      OptionUtils.opt_app
-        (fun x -> "where (" ^ sbt x ^ ")") "" where in
-    Printf.sprintf "update %s set %s %s" table fields where
+  let pr_delete ppf table where =
+    let pp_where ppf where = gen_pp_option ppf "where (%a)" pr_b_ignore_fields where in
+    Format.fprintf ppf "delete from %a%a" pp_constant table pp_where where
```

Comment:
If the table name is foo, this will generate `"delete from foowhere ...`.

Reply:
It's a little worrying that this wasn't picked up by the CI...